File: maths/cycle1/fiches-methodes3e/I/Pascal/m0203.pas
program m0203;

{$mode objfpc}{$H+}

uses
    method02, display;

var
    quotient, remainder, divisor, dividend : Integer;
    bool : Boolean;

begin
    WriteLn('# Exercice 3.');
    WriteLn('# ----------');
    WriteLn('# Dans une division euclidienne, le quotient est 19, le reste est 4 et le');
    WriteLn('# diviseur 6. Quel est le dividende ?');
    WriteLn('-----------------------------------------------------------------------------------');

    bool := DisplayMethodWording;
    if bool then PutAMethod();

    bool := PutUpCorrectVersion;
    if not bool then exit;

    quotient := 19;
    remainder := 4;
    divisor := 6;
    dividend := divisor * quotient + remainder;
    WriteLn('Le dividende est donc ', divisor, ' x ', quotient, ' + ', remainder, ' = ', dividend);
end.

File: maths/cycle1/fiches-methodes3e/I/Ruby/m0203.rb
#!/usr/bin/env ruby

require_relative "method02"
require_relative "corrective"

class Solution < Method02
  include Corrective

  def answering
    quotient = 19
    remainder = 4
    divisor = 6
    dividend = divisor * quotient + remainder
    puts "Le dividende est donc #{divisor} x #{quotient} + #{remainder} = #{dividend}."
  end
end

s = Solution.new
s.exercise_03
bool = true
bool = s.put_up_method?
if bool then s.put_up_method end
s.put_up_correct_version?
s.answering

